pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java

```diff
@@ -197,7 +197,7 @@ public class ModularLoadManagerImpl implements ModularLoadManager {
     private long unloadBundleCount = 0;
 
     private final Lock lock = new ReentrantLock();
-    private Set<String> knownBrokers = new HashSet<>();
+    private Set<String> knownBrokers = ConcurrentHashMap.newKeySet();
```

Comment:
Maybe we can move this line below to method `start` and add some code comments, this could solves the concurrency problem. 


https://github.com/apache/pulsar/blob/7650612cd86ead01886300f9b31aba6946831736/pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java#L249

Reply:
`handleDataNotification ` also has other actions, so we'd better keep it in `init` method I think @poorbarcode